cnxauthoring/storage/postgresql.py

```diff
@@ -328,11 +329,20 @@ def update(self, item_or_items):
 
     def persist(self):
         """Persist/commit the changes."""
-        self.conn.commit()
+        # A try/except statement had been added to handle
+        # situations were the database connection is lost but not caught before
+        # a call to this method (see unittest test_functional.
+        # FunctionalTests.test_storage_unavailable_error). Would suggest
+        # moving all error catching inside this class in the future.
+        try:
+            self.conn.commit()
+        except self.Error:
+            raise httpexceptions.HTTPServiceUnavailable
```

Comment:
this is part of why in my comments I suggested reworking our error catching all together since.  But I think I might have a better solution which I'll push soon.